src/js/components/Button/doc.js

```diff
@@ -44,6 +45,23 @@ export const doc = Button => {
     as: PropTypes.oneOfType([PropTypes.string, PropTypes.func]).description(
       `The DOM tag or react component to use for the element.`,
     ),
+    badge: PropTypes.oneOfType([
+      PropTypes.bool,
+      PropTypes.element,
+      PropTypes.number,
+      PropTypes.shape({
+        background: backgroundPropType,
+        max: PropTypes.number,
+        value: PropTypes.oneOfType([PropTypes.bool, PropTypes.number]),
+      }),
+    ])
+      .description(
+        `An indicator to show on the top-right of the Button. For accessibility,
+         supplement the badge by adding an a11yTitle to the Button which 
+         provides a useful message to screen readers. For example, "2 unread 
+         notifications".`,
```

Comment:
From this description, it is not clear to me what is the behavior for the different types. 
i.e. what is the key/value meaning of the badge object, such as `max`.